ClientApp/viking-vault/src/display_user_list/AddMoneyModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import './AddMoney.css';
+
+interface IAddMoneyModalProps{
+    open: boolean;
+    userName: string;
+    closeModal: any;
+    addMoney: any;
+}
+
+interface IAddMoneyModalState{
+    amount: string;
+}
+
+class AddMoneyModal extends React.Component<IAddMoneyModalProps, IAddMoneyModalState>{ 
+    state = {
+            amount : ""
+        };
+
+    private resetAmountField(){
+        this.setState({
+            amount : ""
+        });
+    }
+    private closeModal = () =>{
+        this.resetAmountField();
+        this.props.closeModal(false);
+    }
+
+    private handleAddMoney = () =>{
+        this.resetAmountField();
+        this.props.addMoney(Number(this.state.amount));
+    }
+
+    private handleAmountChange = (e:any) =>{
+        this.setState({
+            amount: e.target.value
+        });
+    }
+
+    private validateAmount(){
+        if(Number(this.state.amount) <= 0 || Number(this.state.amount) > 5000 || !(this.state.amount.match("^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$"))){
+            return false;
+        }
+        return true;
+    }
+
+    render(){
+        return (
+            <div className={this.props.open? "modal open" : "modal close"}>
+                <div className="modal-dialog modal-dialog-centered">
+                    <div className="modal-content">
+                        <div className="modal-header">
+                            <h4 className="modal-title">Add money to {this.props.userName}</h4>
+                        </div>
+
+                        <div className = "modal-body">
+                            <div className="form-group">
+                                <label>Amount (RON)</label>
+                                <input type="text" className="form-control accent-color" value = {this.state.amount} onChange = {this.handleAmountChange}></input>
+                                {!this.validateAmount() ? <p className="alert alert-danger">Amount should be between 0.1-5000 RON</p> : null}
```

Comment:
Small improvement, but you could extract the minimum amount and the maximum amount in their own separate variables and then use them in here as well as in the `validateAmount`.